src/Sylius/Bundle/ReviewBundle/DependencyInjection/Compiler/RegisterReviewFactoryPass.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ReviewBundle\DependencyInjection\Compiler;
+
+use Sylius\Component\Review\Factory\ReviewFactory;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class RegisterReviewFactoryPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->getParameter('sylius.review.subjects') as $subject => $configuration) {
+            $factory = $container->getDefinition('sylius.factory.'.$subject.'_review');
```

Comment:
Let's use `findDefinition` instead of `getDefinition` to handle aliases correctly.